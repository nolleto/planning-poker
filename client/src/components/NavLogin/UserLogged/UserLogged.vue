<template>
  <div :class="bootstrap.dropdown">
    <a
      :class="[bootstrap.navLink, bootstrap.dropdownToggle]"
      :aria-expanded="ui.expanded"
      id="loginDropdown"
      role="button"
      aria-haspopup="true"
      @click="ui.expanded = !ui.expanded"
    >
      Hello <b>{{ username }}</b>!
    </a>

    <div
      :class="[
        bootstrap.dropdownMenu,
        bootstrap.dropdownMenuRight,
        {
          [bootstrap.show]: ui.expanded
        }
      ]"
      aria-labelledby="loginDropdown"
    >
      <a
        :class="bootstrap.dropdownItem"
        @click="$emit('sign-out')"
      >
        Sign out
      </a>
    </div>
  </div>
</template>

<script>
import styles from '@/styles'

export default {
  name: 'UserLogged',

  props: {
    username: {
      type: String,
      required: true
    }
  },

  data () {
    return {
      ui: {
        expanded: false
      }
    }
  },

  computed: {
    bootstrap: () => styles.bootstrap
  }
}
</script>
