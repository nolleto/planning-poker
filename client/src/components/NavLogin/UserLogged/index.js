export { default } from './UserLogged.vue'
