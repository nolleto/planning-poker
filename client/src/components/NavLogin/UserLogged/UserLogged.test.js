import UserLogged from './index'
import { shallowMount } from '@vue/test-utils'

const defaultProps = {
  username: 'Mauricio da Silva'
}

const shallowComponent = (opts = { propsData: defaultProps }) => {
  return shallowMount(UserLogged, opts)
}

describe('UserLogged component', () => {
  let wrapper

  describe('Snapshots', () => {
    beforeAll(() => {
      wrapper = shallowComponent()
    })

    describe('When is not expanded', () => {
      beforeAll(() => {
        wrapper.setData({ ui: { expanded: false } })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })

    describe('When is expanded', () => {
      beforeAll(() => {
        wrapper.setData({ ui: { expanded: true } })
      })

      it('matches', () => {
        expect(wrapper.element).toMatchSnapshot()
      })
    })
  })

  describe('Event Listeners', () => {
    beforeAll(() => {
      wrapper = shallowComponent()
    })

    describe('When clicked in "Sign out" link', () => {
      beforeEach(() => {
        jest.spyOn(wrapper.vm, '$emit')
        wrapper.findAll('a').at(1).trigger('click')
      })

      it('emits "sign-out"', () => {
        expect(wrapper.vm.$emit).toHaveBeenCalledWith('sign-out')
      })
    })
  })
})
