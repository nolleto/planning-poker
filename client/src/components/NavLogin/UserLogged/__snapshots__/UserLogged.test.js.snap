// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLogged component Snapshots When is expanded matches 1`] = `
<div
  class="dropdown"
>
  <a
    aria-expanded="true"
    aria-haspopup="true"
    class="navLink dropdownToggle"
    id="loginDropdown"
    role="button"
  >
    
    Hello 
    <b>
      Mauricio da Silva
    </b>
    !
  
  </a>
   
  <div
    aria-labelledby="loginDropdown"
    class="dropdownMenu dropdownMenuRight show"
  >
    <a
      class="dropdownItem"
    >
      
      Sign out
    
    </a>
  </div>
</div>
`;

exports[`UserLogged component Snapshots When is not expanded matches 1`] = `
<div
  class="dropdown"
>
  <a
    aria-haspopup="true"
    class="navLink dropdownToggle"
    id="loginDropdown"
    role="button"
  >
    
    Hello 
    <b>
      Mauricio da Silva
    </b>
    !
  
  </a>
   
  <div
    aria-labelledby="loginDropdown"
    class="dropdownMenu dropdownMenuRight"
  >
    <a
      class="dropdownItem"
    >
      
      Sign out
    
    </a>
  </div>
</div>
`;
